omc3/scripts/luminosity_imbalance.py

```diff
@@ -0,0 +1,118 @@
+
+
+def _validate_tfs(df):
+    # First validate the labels, which correspond to the IP and Beam
+    expected_labels = set(('IP1B1', 'IP1B2', 'IP5B1', 'IP5B2'))
+    if set(df['LABEL']) != expected_labels:
+        LOG.error("Expected labels in the TFS file not found. Expected labels:"
+                  " {}".format(expected_labels))
+        raise KeyError('Labels not found in TFS file')
+
+    # Validate the columns we need: BETSTAR{X,Y} and ERRBETSTAR{X,Y}
+    expected_columns = ('BETSTARX', 'BETSTARY', 'ERRBETSTARX', 'ERRBETSTARY')
+    if not all([column in df.columns for column in expected_columns]):
+        LOG.error("Expected columns in the TFS file not found. Expected "
+                  "columns: {}".format(expected_columns))
+        raise KeyError('Columns not found in TFS file')
+
+
+def get_imbalance(df):
+    # Get the individual rows for the IP1 and IP5
+    ip1 = df.loc[df['LABEL'].str.startswith('IP1')]
+    ip5 = df.loc[df['LABEL'].str.startswith('IP5')]
+
+    # Compute the average β* at each IP and plane with both beams
+    average_x_ip1 = ip1.sum(axis=0)['BETSTARX'] / 2
+    average_x_ip5 = ip5.sum(axis=0)['BETSTARX'] / 2
+    average_y_ip1 = ip1.sum(axis=0)['BETSTARY'] / 2
+    average_y_ip5 = ip5.sum(axis=0)['BETSTARY'] / 2
+
+    # Compute the luminosity at each IP
+    eff_beta_ip1 = 1 / sqrt(1 / (average_x_ip1 * average_y_ip1))
+    eff_beta_ip5 = 1 / sqrt(1 / (average_x_ip5 * average_y_ip5))
+
+    # And get the imbalance
+    imbalance = eff_beta_ip1 / eff_beta_ip5
+
+    # Compute the absolute errors for each IP and plane
+    error_x_ip1 = 0.5 * sqrt((ip1['ERRBETSTARX'] ** 2).sum(axis=0))
+    error_y_ip1 = 0.5 * sqrt((ip1['ERRBETSTARY'] ** 2).sum(axis=0))
+    error_x_ip5 = 0.5 * sqrt((ip5['ERRBETSTARX'] ** 2).sum(axis=0))
+    error_y_ip5 = 0.5 * sqrt((ip5['ERRBETSTARY'] ** 2).sum(axis=0))
+
+    # Compute the relative errors for each IP
+    rel_error_ip1 = 0.5 * sqrt((error_x_ip1 ** 2) / (average_x_ip1 ** 2)
+                               + (error_y_ip1 ** 2) / (average_y_ip1 ** 2))
+    rel_error_ip5 = 0.5 * sqrt((error_x_ip5 ** 2) / (average_x_ip5 ** 2)
+                               + (error_y_ip5 ** 2) / (average_y_ip5 ** 2))
+    relative_error = rel_error_ip1 + rel_error_ip5
+
+    return {'imbalance': imbalance, 
+            'relative_error': relative_error,
+            'eff_beta_ip1': eff_beta_ip1, 
+            'rel_error_ip1': rel_error_ip1,
+            'eff_beta_ip5': eff_beta_ip5, 
+            'rel_error_ip5': rel_error_ip5
+            }
+
+
+def print_luminosity(res):
+    print('{:22s}'.format("Luminosity imbalance:"), res['imbalance'])
+    print('{:22s}'.format("Relative error:"), res['relative_error'])
+    print('{:22s}'.format("Effective beta IP1:"), res['eff_beta_ip1'])
+    print('{:22s}'.format("Rel error IP1:"), res['rel_error_ip1'])
+    print('{:22s}'.format("Effective beta IP5:"), res['eff_beta_ip5'])
+    print('{:22s}'.format("Rel error IP5:"), res['rel_error_ip5'])
```

Comment:
Lukas will complain that it's not f strings .. 
also maybe using tfs.tools. significant digits could be used to shorten the output

Reply:
I didn't get the part about significant digits, what's the desired output?

Reply:
so instead of having an output of 0.123456789 +- 0.00123456 , printing just 0.123 +- 0.001 would be more sensible .. since we only measure with the precision of the error, any digits in the actual result would not be particularly meaningful
https://en.wikipedia.org/wiki/Significant_figures

Reply:
I understand what significant digits are but why would you want to shorten the output?

Reply:
readability